chaincode/src/allowances/grantAllowance.spec.ts

```diff
@@ -201,6 +197,50 @@ describe("GrantAllowance", () => {
     expect(writes).toEqual(writesMap(allowance));
   });
 
+  it("should allow infinite Allowances for fungible token", async () => {
+    // Given
+    const currencyInstance = currency.tokenInstance();
+    const currencyClass = currency.tokenClass();
+    const currencyClassKey = currency.tokenClassKey();
+    const currencyInstanceQueryKey = await createValidDTO(
+      TokenInstanceQueryKey,
+      currency.tokenInstanceKeyPlain()
+    );
+
+    const currencyBalance = plainToInstance(TokenBalance, {
+      owner: users.testUser2Id,
+      ...currencyClassKey,
+      instanceIds: [],
+      quantity: new BigNumber("1000")
+    });
+
+    const { ctx, contract, writes } = fixture(GalaChainTokenContract)
+      .callingUser(users.testUser2Id)
+      .savedState(currencyClass, currencyInstance, currencyBalance);
+
+    const dto: GrantAllowanceDto = await createValidDTO(GrantAllowanceDto, {
+      tokenInstance: currencyInstanceQueryKey,
+      quantities: [{ user: users.testUser1Id, quantity: new BigNumber(Infinity) }],
+      allowanceType: AllowanceType.Lock,
+      uses: new BigNumber("1")
+    });
+
+    // When
+    const response = await contract.GrantAllowance(ctx, dto);
+
+    // Then
+    const allowance = currency.tokenAllowance((a) => ({
+      ...a,
+      created: ctx.txUnixTime,
+      quantity: new BigNumber(Infinity),
+      grantedBy: users.testUser2Id,
+      grantedTo: users.testUser1Id,
+      allowanceType: AllowanceType.Lock
+    }));
+    expect(response).toEqual(GalaChainResponse.Success([allowance]));
+    expect(writes).toEqual(writesMap(allowance));
+  });
+
```

Comment:
I think there should be also tests that verify mint allowances:
- should fail to grant mint allowance for fungible token when quantity is greater than balance
- should fail to grant infinite mint allowance

Reply:
The first one doesn't really make sense I don't think since mint allowances are granted by token authorities. So there isn't really a balance to compare against. But, the 2nd one is simple enough